extensions/resource-deployment/src/ui/modelViewUtils.ts

```diff
@@ -408,6 +409,58 @@ async function hookUpDynamicEnablement(context: WizardPageContext): Promise<void
 	}));
 }
 
+/**
+ * Hooks up the dynamic options for fields which use that. This will attach a listener to the target component
+ * for when the value changes and update the options of the source component based on the current value
+ * of the target component.
+ *
+ * Note that currently this is only supported for Notebook Wizard Pages and only supports direct equals comparison
+ * for the value currently selected.
+ *
+ * Additionally this only supports hooking up components that are on the same page.
+ * @param context The page context
+ */
+async function hookUpDynamicOptions(context: WizardPageContext): Promise<void> {
+	await Promise.all(context.pageInfo.sections.map(async section => {
+		if (!section.fields) {
+			return;
+		}
+		await Promise.all(section.fields.map(async field => {
+			if (instanceOfDynamicOptionsInfo(field.dynamicOptions)) {
+				const fieldKey = field.variableName || field.label;
+				const fieldComponent = context.inputComponents[fieldKey];
+				const targetComponent = context.inputComponents[field.dynamicOptions.target];
+				if (!targetComponent) {
+					console.error(`Could not find target component ${field.dynamicOptions.target} when hooking up dynamic options for ${field.label}`);
+					return;
+				}
+				const updateOptions = async () => {
+					const currentValue = await targetComponent.getValue();
+					if (field.dynamicOptions && field.options && fieldComponent) {
+						const targetValueFound = field.dynamicOptions.alternates.find(item => item.selection === currentValue);
+						if (targetValueFound) {
+							(<RadioGroupLoadingComponentBuilder>fieldComponent.component).loadOptions(<OptionsInfo>{
```

Comment:
This is currently only going to work for radio groups it seems. What would be nice is to make a `setOptions` or something on the generic InputComponentInfo that each component could then implement as they need (so radio button groups would call loadOptions like you do here). This way we can easily add support for more types of components in the future.